modules/python/nitf/source/nitro.i

```diff
@@ -156,6 +156,17 @@
     }
 }
 
+%typemap(in) nrt_IOHandle{
+    if ($input)
+    {
+        $1 = (nrt_IOHandle) PyInt_AsLong($input);
+    }
+}
+
+%typemap(out) nrt_IOHandle{
+    $result = SWIG_From_int((int)($1));
```

Comment:
It looks like PyInt_AsSsize_t returns a signed type with the size of size_t.

What was going on here is that SWIG was auto-generating something like arg1 = SWIG_From_int(static_cast<int>(ioHandle)), which wasn’t compiling on Windows.

Anyway, I’ve changed things so that it should work with the typedefs on both platforms.

From: Adam Sylvester [mailto:notifications@github.com]
Sent: Thursday, August 25, 2016 10:04 PM
To: mdaus/nitro <nitro@noreply.github.com>
Cc: Jonathan Means <Jonathan.Means@mdaus.com>; Author <author@noreply.github.com>
Subject: Re: [mdaus/nitro] Fix compilation and run-time errors on Windows (#21)


In modules/python/nitf/source/nitro.i<https://github.com/mdaus/nitro/pull/21#discussion_r76354670>:

> @@ -156,6 +156,17 @@

>      }

>  }

>

> +%typemap(in) nrt_IOHandle{

> +    if ($input)

> +    {

> +        $1 = (nrt_IOHandle) PyInt_AsLong($input);

> +    }

> +}

> +

> +%typemap(out) nrt_IOHandle{

> +    $result = SWIG_From_int((int)($1));

Similarly here... I'm not familiar with this syntax but do you need PyInt_FromSize_t()?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/mdaus/nitro/pull/21/files/1069ab16a6548aba3001856648ee6fef72d0bbc5#r76354670>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AFRAYL7Of0MMbSV8M9TDK-pWSvmIB9jzks5qjkmkgaJpZM4JtGwL>.


The information contained in this communication is confidential, is intended only for the use of the recipient(s) named above, and may be legally privileged. If the reader of this message is not the intended recipient, you are hereby notified that any dissemination, distribution, or copying of this communication is strictly prohibited. If you have received this communication in error, please re-send this communication to the sender and delete the original message or any copy of it from your computer system.